Source/inv.cpp

```diff
@@ -2045,6 +2045,20 @@ bool UseInvItem(int pnum, int cii)
 				break;
 			}
 		}
+
+		// If speedlist item is not inventory, use same item at the end of the speedlist if exists.
+		if (speedlist) {
+			for (int i = INVITEM_BELT_LAST - INVITEM_BELT_FIRST; i > cii - INVITEM_BELT_FIRST && *sgOptions.Gameplay.autoRefillBelt; i--) {
+				Item candidate = player.SpdList[i];
+
+				if (candidate._iMiscId == item->_iMiscId && candidate._iSpell == item->_iSpell && candidate._itype != ItemType::None) {
+					c = i;
+					item = &player.SpdList[c];

```

Comment:
```suggestion
					item = &candidate;
```

No need to use the array index again here.